modules/openapi-generator/src/test/resources/3_0/schema-with-special-characters.yaml

```diff
@@ -0,0 +1,59 @@
+openapi: 3.0.3
+info:
+  description: test
+  version: 1.0.0
+  title: test
+paths:
+  /test:
+    post:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/MySchemaName._-Characters'
+      responses:
+        '200':
+          description: the response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MySchemaName._-Characters'
+components:
+  schemas:
+    Parent:
+      properties:
+        prop1:
+          type: string
+      discriminator:
+        propertyName: prop1
+    # Note: the name below with non-ASCII characters is a valid JSON schema name,
+    # however currently OpenAPI generator does not allow special characters beyond [.-_]
+    #
+    # The set of allowed characters for OpenAPI schema names is specified in RFC 8259
+    # at https://tools.ietf.org/html/rfc8259.
+    # The OpenAPI schema uses the JSON schema specification, which references RFC 8259.
+    # A string is a sequence of zero or more Unicode characters [UNICODE].
+    # Note that this citation references the latest version of Unicode
+    # rather than a specific release. Any character may be escaped.
+    # MySchemaNameWithUnusual#$12.3!@#%🍇🍅😀🏄🏾‍♂️Characters:
```

Comment:
@wing328 , `MySchemaNameWithUnusual#$12.3!@#%🍇🍅😀🏄🏾‍♂️Characters` is a valid JSON schema name, but OpenAPI generator does not allow this.